### PR TITLE
update some sources according new if_gen.py script

### PR DESCRIPTION
**Pull request by JDLopes**

### CODE REVIEW

File: hardware/include/hw_setup.mk [hunk 1]
```diff
@@ -10,4 +10,18 @@ SRC+=$(subst $(LIB_DIR)/hardware/include, $(BUILD_VSRC_DIR), $(wildcard $(LIB_DI
...
+iob_s_port.vh:
+	$(LIB_DIR)/scripts/if_gen.py iob_s_port '' ''
+
+# iob slave portmap for swreg files
+SRC+=$(BUILD_VSRC_DIR)/iob_s_portmap.vh
+$(BUILD_VSRC_DIR)/iob_s_portmap.vh: iob_s_portmap.vh
+	cp $< $@
+iob_s_portmap.vh:
+	$(LIB_DIR)/scripts/if_gen.py iob_s_portmap '' ''
+
```
Comment (jjts) [inline hardware/include/hw_setup.mk:26]: Not sure we need this ...

Now that we have the script we want to generate iob_ interfaces with our prefixes and not use generic ones 

include include/hw_setup imply that we want to generate 

For directory include

mv axi_tasks.vh ../axil2iob/axil_tasks.vh

and then edit  ../axil2iob/hw_setup to move axil_tasks.vh to BUILD_SIM_DIR

Include the contents of iob_intercon.vh in iob_lib.vh and delete  iob_intercon.vh

Most likely this contents will become obsolete as, with the script it is no longer a pain to generate the iob_interface signals


the  contents of iob_intercon.vh can be deleted when we update iob_merge and iob_split to use the if_gen.py, CPU wrappers and cache

Comment (JDLopes) [reply hardware/include/hw_setup.mk:26]: > Now that we have the script we want to generate iob_ interfaces with our prefixes and not use generic ones
Those are there for the ones required by the files generated by mkregs.py script. If one has a project with many swreg files for many cores, only two files are needed, and the user does not need to worry about them.

Comment (jjts) [reply hardware/include/hw_setup.mk:26]: add 

ifneq ($(wildcard mkregs.conf),)

then

Comment (JDLopes) [reply hardware/include/hw_setup.mk:28]: Done!

**Comment by jjts** [comment]
name of the file should be axil_tasks.vh and not axi_tasks.vh

**Pull request merged**